src/Npgsql/NpgsqlSchema.cs

```diff
@@ -438,6 +439,291 @@ public static DataTable GetReservedWords()
+            var obj = new object[40, 23] {
+            {"CHAR", (int)NpgsqlDbType.Char, DBNull.Value, "CHAR({0})",
+                    "length", "System.String",
+                    false, false, true, true, false,
+                    false, true, true, true, DBNull.Value,
+                    DBNull.Value, DBNull.Value,
+                    false, true, "'", "'", "DBTYPE_STR"},
+            {"VACHAR",(int)NpgsqlDbType.Varchar, DBNull.Value, "VARCHAR({0})",
+                    "max length", "System.String",
+                    false, false, true, false, false,
+                    false, true, true, true, DBNull.Value,
+                    DBNull.Value, DBNull.Value,
+                    false, true, "'", "'", "DBTYPE_STR"},
+            {"NUMERIC",(int)NpgsqlDbType.Numeric, DBNull.Value, "NUMERIC({0}, {1})",
+                    "precision,scale", "System.Decimal",
+                    false, false, false, false, false,
+                    false, true, true, false, false,
+                    1000, 0, false, true,
+                    DBNull.Value, DBNull.Value, "DBTYPE_DECIMAL"},
+            {"BIGINT", (int)NpgsqlDbType.Bigint, 8, "BIGINT",
+                    DBNull.Value, "System.Int64",
+                    true, true, false, true, true,
+                    false, true, true, false, false,
+                    DBNull.Value, DBNull.Value, false, true,
+                    DBNull.Value, DBNull.Value, "DBTYPE_I8"},
+            {"INTEGER",(int)NpgsqlDbType.Integer, 4, "INTEGER",
+                    DBNull.Value, "System.Int32",
+                    true, true, false, true, true,
+                    false, true, true, false, false,
+                    DBNull.Value, DBNull.Value, false, true,
+                    DBNull.Value, DBNull.Value, "DBTYPE_I4"},
+            {"SMALLINT",(int)NpgsqlDbType.Smallint, 2, "SMALLINT",
+                    DBNull.Value, "System.Int16",
+                    true, true, false, true, true,
+                    false, true, true, false, false,
+                    DBNull.Value, DBNull.Value, false, true,
+                    DBNull.Value, DBNull.Value, "DBTYPE_I2"},
+            {"REAL",(int)NpgsqlDbType.Real, 4, "REAL",
+                    DBNull.Value, "System.Float",
+                    false, true, false, true, false,
+                    false, true, true, false, false,
+                    DBNull.Value, DBNull.Value, false, true,
+                    DBNull.Value, DBNull.Value, "DBTYPE_R4"},
+            {"DOUBLE", (int)NpgsqlDbType.Double, 8, "DOUBLE",
+                    DBNull.Value, "System.Double",
+                    false, true, false, true, false,
+                    false, true, true, false, false,
+                    DBNull.Value, DBNull.Value, false, true,
+                    DBNull.Value, DBNull.Value, "DBTYPE_R8"},
+            {"DATE", (int)NpgsqlDbType.Date, 4, "DATE",
+                    DBNull.Value, "System.DateTime",
+                    false, true, false, true, true,
```

Comment:
Should have `IsBestMatch` false, `TIMESTAMP` below is a better match for .NET DateTime...